docs/proposals/31-inplace-node-update.md

```diff
@@ -0,0 +1,441 @@
+      cpu: "2"
+      gpu: "0"
+      memory: 8Gi
+      usable: true
+  regions:
+    - name: <region-name>
+```
+
+#### `Shoot` API
+
+A new field `updateStrategy` is introduced under `spec.provider.workers[]` in the Shoot spec. This field will be passed on to the worker extension. Once set, changing the `updateStrategy` from an in-place strategy to a replace strategy and vice versa is prohibited through validation. Additionally, when `AutoInPlaceUpdate` or `ManualInPlaceUpdate` is configured as an update strategy, skipping an intermediate Kubernetes minor version is no longer allowed, because of potential breaking changes that are then not carried out anymore when skipping versions (only allowed with `AutoReplaceUpdate` as the machines/nodes are created from scratch in this case).
+
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: Shoot
+metadata:
+  name: crazy-botany
+  namespace: garden-dev
+spec:
+  secretBindingName: my-provider-account
+  cloudProfile:
+    name: cloudprofile1
+  region: europe-central-1
+  provider:
+    type: <some-provider-name> # {aws,azure,gcp,...}
+    workers:
+      - name: cpu-worker
+        minimum: 5
+        maximum: 5
+        maxSurge: 0
+        maxUnavailable: 2
+        updateStrategy: AutoInPlaceUpdate # AutoReplaceUpdate/AutoInPlaceUpdate/ManualInPlaceUpdate, defaulted for now to AutoReplaceUpdate
+        machine:
+          type: m5.large
+          image:
+            name: <some-image-name>
+            version: <some-image-version>
+          architecture: <some-cpu-architecture>
+        providerConfig: <some-machine-image-specific-configuration>
+  kubernetes:
+    version: 1.27.3
+  networking:
+    type: <some-network-extension-name> # {calico,cilium}
+```
+
+#### `Worker` API
+
+New field `updateStrategy` and `kubelet` are introduced under `.spec.pools[]` in the Worker API. The `updateStrategy` field will be used to set the `machineDeploymentStrategy` for the `machinev1alpha1.MachineDeployment` resource and `kubelet` field corresponds to the kubelet configuration for this worker pool.
+
+```yaml
+apiVersion: extensions.gardener.cloud/v1alpha1
+kind: Worker
+metadata:
+  name: worker
+  namespace: shoot--foobar--aws
+spec:
+  type: aws
+  region: eu-west-1
+  secretRef:
+    name: cloudprovider
+    namespace: shoot--foobar--aws
+  infrastructureProviderStatus: <infra-status>
+  pools:
+    - name: cpu-worker-1
+      machineImage:
+        name: gardenlinux
+        version: 1.2.2
+      minimum: 5
+      maximum: 5
+      maxSurge: 0
+      maxUnavailable: 2
+      updateStrategy: AutoInPlaceUpdate # AutoReplaceUpdate/AutoInPlaceUpdate/ManualInPlaceUpdate
+      kubelet:
+        evictionHard:
+          memoryAvailable: 100Mi
+          imageFSAvailable: 5%
+```
+
+#### `OperatingSystemConfig` API
+
+New fields `osVersion`, `kubeletVersion`, and `credentialsRotation` are introduced in the `OperatingSystemConfig` spec and `inPlaceUpdateConfig` in the `OperatingSystemConfig` status.
```

Comment:
OperatingSystemConfig is no longer just OS config, right? Are there are any plans to either split or rename it?